### ISSUE TITLE
Far too many branches

### ISSUE THREAD

**Issue by StephanEggermont** [pharo-ide/Calypso#473]
Contributing to Calypso is difficult when it is not clear which branch to push to. Please reduce to the absolute minimum

**Comment by jecisc** [comment]
Better would maybe be to have a CONTRIBUTION.md file?

**Comment by dionisiydk** [comment]
Did you see bold section "Contribution (Pharo 7 and higher)" in readme?

**Comment by StephanEggermont** [comment]
No. Why is there a dev branch?

**Comment by dionisiydk** [comment]
All development is accumulated in dev branch through PRs. When we decide that it is time for release we push dev to master and create a new release tag. In addition we push (or create) to floating branch (0.16.x for example).
This way master is always stable.

You can propose better process. But honestly I don't have time to spend on it.

**Comment by StephanEggermont** [comment]
Drop dev, do that all in master. No branch that adds no value. Currently, master doesn't.

**Comment by dionisiydk** [comment]
I think it has value.
For example loading master gives you latest stable release.

**Comment by StephanEggermont** [comment]
No, because there is no such thing. Release for what? Stable is different for 7 and for 8.

**Comment by dionisiydk** [comment]
I wonder about your No. It sounds like I ask you how it works and you
explain me how. But it seems I am explaining you how it is done :)

Stable is same for Pharo 7 and 8. At least for now.
But in general you can consider master as a latest stable version for
supported Pharo versions.

вс, 24 мар. 2019 г. в 19:42, Stephan Eggermont <notifications@github.com>:

**Comment by StephanEggermont** [comment]
Sorry, that’s because I’ve been fighting bad dependency management in Pharo for the past five years or so, and this exact problem cost a client of mine about 30 man months two years ago.

**Comment by jecisc** [comment]
I work with master and dev branch on projects that have more than 75 dependencies without problem from that kind of management. For me the problem does not comes from there.

**Comment by StephanEggermont** [comment]
It just creates extra work and confusion. The absolute number of dependencies is only a small part of it.
